chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/iobuffer/FieldMultiArrayHelper.java

```diff
@@ -0,0 +1,182 @@
+            final int nElements = serialiser.putArraySizeDescriptor(value.getDimensions());
+            serialiser.getBuffer().putByteArray(((MultiArrayByte) value).elements(), nElements);
+            serialiser.updateDataEndMarker(fieldHeader);
+        } else if (value instanceof MultiArrayObject) {
+            final WireDataFieldDescription fieldHeader = serialiser.putFieldHeader(fieldName, DataType.STRING_ARRAY);
+            final int nElements = serialiser.putArraySizeDescriptor(value.getDimensions());
+            serialiser.getBuffer().putStringArray(((MultiArrayObject<String>) value).elements(), nElements);
+            serialiser.updateDataEndMarker(fieldHeader);
+        } else {
+            throw new IllegalArgumentException("Illegal DataType for MultiArray");
+        }
+    }
+
+    public static <E> void put(final IoSerialiser serialiser, final FieldDescription fieldDescription, final MultiArray<E> value) {
+        final WireDataFieldDescription fieldHeader = serialiser.putFieldHeader(fieldDescription);
+        final int nElements = serialiser.putArraySizeDescriptor(value.getDimensions());
+        switch (fieldDescription.getDataType()) {
+        case BOOL_ARRAY:
+            serialiser.getBuffer().putBooleanArray(((MultiArrayBoolean) value).elements(), nElements);
+            break;
+        case BYTE_ARRAY:
+            serialiser.getBuffer().putByteArray(((MultiArrayByte) value).elements(), nElements);
+            break;
+        case SHORT_ARRAY:
+            serialiser.getBuffer().putShortArray(((MultiArrayShort) value).elements(), nElements);
+            break;
+        case INT_ARRAY:
+            serialiser.getBuffer().putIntArray(((MultiArrayInt) value).elements(), nElements);
+            break;
+        case LONG_ARRAY:
+            serialiser.getBuffer().putLongArray(((MultiArrayLong) value).elements(), nElements);
+            break;
+        case FLOAT_ARRAY:
+            serialiser.getBuffer().putFloatArray(((MultiArrayFloat) value).elements(), nElements);
+            break;
+        case DOUBLE_ARRAY:
+            serialiser.getBuffer().putDoubleArray(((MultiArrayDouble) value).elements(), nElements);
+            break;
+        case CHAR_ARRAY:
+            serialiser.getBuffer().putCharArray(((MultiArrayChar) value).elements(), nElements);
+            break;
+        case STRING_ARRAY:
+            serialiser.getBuffer().putStringArray(((MultiArrayObject<String>) value).elements(), nElements);
+            break;
+        default:
+            throw new IllegalStateException("Unexpected value: " + fieldDescription.getDataType());
+        }
+        serialiser.updateDataEndMarker(fieldHeader);
+    }
+
+    /**
+     * Registers default serialiser for MultiArray
+     *
+     * @param serialiser for which the field serialisers should be registered
+     */
+    public static void register(final IoClassSerialiser serialiser) {
```

Comment:
Codacy found an issue: [The method 'register(IoClassSerialiser)' has an NPath complexity of 512, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)